File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    # 완전탐색 
    # 중간에 삽입
    
    # 각 뭉치의 다음 순서에 나올 것을 저장, 비교해서 둘 다 아니면 no
    # 둘 중 하나면 다음 순서에 나올 것 갱신
    # goal을 순회하면서 cards1, cards2에 나올 다음 단어를 저장하기 
    next1, next2 = 0,0
    len1 = len(cards1)
    len2 = len(cards2)
    for word in goal:
        if next1 < len1 and word == cards1[next1]:
            next1 += 1
        elif next2 < len2 and word == cards2[next2]:
            next2 += 1
        else:
            return "No"
    return "Yes"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public static void log(Object o){
        System.out.println(o);
    }
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        /*
        다음 찾는 단어가 양쪽에 있는지 확인
        있으면 인덱스 갱신
        없으면 No 리턴
        다 돌면 Yes
        */
        
        int i = 0;
        int j = 0;
        int n = cards1.length;
        int m = cards2.length;
        for(String word: goal){
            //log(word +" " + cards1[i] + " " + cards2[j]);
            if(i < n && cards1[i].equals(word)){
                i++;
                continue;
            }
            if(j < m && cards2[j].equals(word)){
                j++;
                continue;
            }
            return "No";
        }
        return "Yes";
    }
}